hudi-common/src/test/java/org/apache/hudi/model/TestHoodieAvroRecordMerger.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.model;
+
+import org.apache.hudi.common.HoodieJsonPayload;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.model.HoodieAvroRecord;
+import org.apache.hudi.common.model.HoodieAvroRecordMerger;
+import org.apache.hudi.common.model.HoodieRecord;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.collection.Pair;
+
+import org.apache.avro.Schema;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.apache.hudi.model.TestUtil.SCHEMA;
+import static org.apache.hudi.model.TestUtil.generateData;
+
+public class TestHoodieAvroRecordMerger {
+  private static final HoodieAvroRecordMerger MERGER = HoodieAvroRecordMerger.INSTANCE;
+
+  @Test
+  public void testMergeWhenBothSidesAreGood() throws IOException {
+    List<HoodieAvroRecord> olderRecords = generateData(10);
+    List<HoodieAvroRecord> newerRecords = generateData(10);
+
+    for (int i = 0; i < olderRecords.size(); ++i) {
+      Option<Pair<HoodieRecord, Schema>> r = MERGER.merge(
+          Option.of(olderRecords.get(i)),
+          SCHEMA,
+          Option.of(newerRecords.get(i)),
+          SCHEMA,
+          new TypedProperties());
+      Assertions.assertEquals(r.get().getRight(), SCHEMA);
+      Assertions.assertEquals(
+          r.get().getLeft().getData(),
+          ((HoodieJsonPayload) newerRecords.get(i).getData()).getInsertValue(SCHEMA).get());
+    }
+  }
+
+  @Test
+  public void testMergeWhenOneSideIsEmpty() throws IOException {
+    List<HoodieAvroRecord> records = generateData(1);
+
+    for (HoodieAvroRecord record : records) {
+      Option<Pair<HoodieRecord, Schema>> r = MERGER.merge(
+          Option.empty(),
```

Comment:
Could you also add test cases where the invalid record is represented as `HoodieEmptyRecord` or `HoodieRecord.SENTINEL` used by `HoodieAvroRecord`?